src/xdsl/parser.py

```diff
@@ -205,6 +205,9 @@ def parse_optional_region(self) -> Optional[Region]:
         if not self.parse_optional_char("{"):
             return None
         region = Region()
+        oldSSAVals = self._ssaValues.copy()
```

Comment:
I am not quite up to speed with the parser anymore. Does this clear the SSAValue dict? It has to right? O/w, you might not catch doubly defined SSAValues. What am I missing?

Reply:
So, this will keep the `SSA Value context` that was at the beginning of the region, and then reset it at the end.
This essentially ensure that the values in the region will not show outside of it!